### PR TITLE
Fix create order customization/combinations

### LINKED ISSUE #17085: Migration - Create order bugs
## What's left

- #17179

## Done / cannot reproduce

**Describe the bug**
Following the merge of https://github.com/PrestaShop/PrestaShop/pull/16207, here is the list of the remaining bugs on this page :

1. Voucher with gift product does not work : The gift should appear in the cart, with price replaced by "Gift", but nothing happens. However, once the order is created, the gift is in the order.

➡️ PR https://github.com/PrestaShop/PrestaShop/pull/17179

2. The combinations of product with combination are no longer loaded immediately on search : 
https://drive.google.com/file/d/1gWseCzlCvy7I4e1ijpRcnlafCaquZD1R/view?usp=sharing

➡️ fixed by https://github.com/PrestaShop/PrestaShop/pull/17465 ✅ 

3. When adding a "Free Shipping" voucher on an order (NOT created by checking the "Free Shipping" on the page), and then, checking the "Free Shipping" option on the order, the Shipping price is deducted twice.
See this 1.7.7.x invoice : 
[Invoice two free ship 177.pdf](https://github.com/PrestaShop/PrestaShop/files/4147295/Invoice.two.free.ship.177.pdf)
And the same action on 1.7.6 (free shipping is only deducted once) : 
[Invoice two free ship 176.pdf](https://github.com/PrestaShop/PrestaShop/files/4147322/Invoice.two.free.ship.176.pdf)

➡️ we failed to reproduce it

4. file name is not displayed in migrated when adding file for a custom product 
screenshot 1.7.7 : 
upload_custom_file_name_1 7 7
in legacy 1.7.6 : 
upload_custom_file_name_1 7 6

Please note that there might still be some bugs out there...

**Additional information**
PrestaShop version: 1.7.7.x
PHP version: N/A

### LINKED ISSUE #17205: Customization field not automatically displayed while creating an order
**Describe the bug**
Customization field not automatically displayed while creating an order

**To Reproduce**
1. Go to BO > order > add new order
2. Select a customer
3. Search a product with customization field like "personnalisable mug"
4. The product is automatically selected but the customization field isn't displayed, it appears if you select another product and this one again

**Screenshots**
capture d'écran_2287

**Additional information**
PrestaShop version: 1.7.7.x
PHP version: N/A

### PR DESCRIPTION
**Pull request by atomiix**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | When adding a product in create order, if the first product on the result list had customizations or combinations, the corresponding field were not displayed
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17205 & #17198 & Partial fix #17085 
| How to test?  | In *Orders > Create order* on the migrated page, search for a product with a customization. The customization fields should appears automatically. Same thing with combinations. => URL : /admin-dev/index.php/sell/orders/orders/new

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17465)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/pages/order/create/product-manager.js [hunk 1]
```diff
@@ -219,8 +219,10 @@ export default class ProductManager {
   _selectFirstResult() {
     this._unsetProduct();
 
-    if (this.products.length !== 0) {
-      this._selectProduct(this.products[0].productId);
+    const keys = Object.keys(this.products);
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/product-manager.js]: Why not taking directly the `Object.values` and looking for the first result? :thinking:

Comment (atomiix) [reply admin-dev/themes/new-theme/js/pages/order/create/product-manager.js]: Yeah, why not? 😅

Comment (atomiix) [reply admin-dev/themes/new-theme/js/pages/order/create/product-manager.js]: Done

**Comment by atomiix** [comment]
As I was working on this page, I also fixed the style on the `select` elements

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by SD1982** [comment]
@atomiix QA approved

**Comment by jolelievre** [comment]
Thank you @atomiix

**Pull request merged**